### DISCUSSION TITLE
msnoise master + plot dvv + specify the pair

### DISCUSSION THREAD

**Discussion by gmsilveira** [ROBelgium/MSNoise#342] [General]
Dear colleagues,

I am using the msnoise master. I am trying to plot the dvv output using 
msnoise cc dvv plot dvv 
and adding the " -p, --pair TEXT          Plot a specific pair" option.
But, I have the feeling that I always get the average of the results for all paths. 

Many thanks for any help.

**Comment by ThomasLecocq** [comment]
Hey, 

The pair-plot option has not been implemented yet, I think! The easiest way to get the values for a single pair, currently, would be to use the API .

Tom

**Reply by gmsilveira** [reply]
Thanks Thomas, I suspected that. 
Cheers
Graça

**Reply by gmsilveira** [reply]
Hi Thomas,
I'm working on that, but the master output is different from the previous versions. 
I have the impression that the output for each pair is in DTT2/filter/mov_stack and that DVV2 has the average of all pairs.
But, maybe I am wrong as my plot_dvv.py keeps telling me that 
Error fetching data: (sqlite3.OperationalError) no such table

Sorry and thanks

**Reply by ThomasLecocq** [reply]
you can use the API to get your results:

```python

from msnoise.api import *
db = connect()
start, end, datelist = build_movstack_datelist(db)

filterid = 1
mov_stack = 1
dttname = "M0"

dvv, edvv = get_dvv(db, filterid, components, dates, mov_stack=mov_stack, aggregation="median", dttname=dttname)
```

(untested, but should be something like that)
note this method, for some stupid reason I have to check, groups the results by DAY! 

Alternatively, re-computing the DVV from the DTT results should work too:

```
params = get_params(db)
pairs = [('XX.YYY.--', 'XX.ZZZ.--'),] # list of tuples of NET.STA.LOC
components = ["ZZ",]
mov_stack = ('1d', '1d')

dvv = compute_dvv(db, filterid, mov_stack, pairs=pairs, components=components, params=params, method=None)
```

**Reply by ThomasLecocq** [reply]
also check this notebook that has example plots for individual pairs/components: 
https://github.com/heinerigel/skience2024/blob/main/03%20MSNoise/03%20-%20Recompute%20and%20subdaily.ipynb

**Reply by gmsilveira** [reply]
Thanks, I managed to made also an alternative version which is working. But I will try your suggestions. 
I had a stupid mistake.
